lib/interfaces/express.js

```diff
@@ -25,66 +25,56 @@ var expressRequestInformationExtractor =
 var errorHandlerRouter = require('../error-router.js');
 
 /**
- * The Express Error Handler function is an interface for the error handler
- * stack into the Express architecture. This function accepts four arguments:
- * a bound reference to a client which should be given by the proceeding
- * handlerSetup function, an error argument which can be of any type, an
- * express request object, an express response object and the express next
- * function for passing input up the plugin chain.
- * @function expressErrorHandler
- * @param {AuthClient} client - a bound and inited Auth Client instance
- * @param {NormalizedConfigurationVariables} config - the environmental
+ * Returns a function that can be used as an express error handling middleware.
+ * @function makeExpressHandler
+ * @param {AuthClient} client - an inited Auth Client instance
+ * @param {NormalizedConfigurationVariables} config  - the environmental
  *  configuration
- * @param {Any} err - a error of some type propagated by the express plugin
- *  stack
- * @param {Object} req - an Express request object
- * @param {Object} res - an Express response object
- * @param {Function} next - an Express continuation callback
- * @returns {ErrorMessage} - Returns the ErrorMessage instance
+ * @returns {expressErrorHandler} - a function that can be used as an express
+ *  error handling middleware.
  */
-function expressErrorHandler(client, config, err, req, res, next) {
-  var ctxService = '';
-  var ctxVersion = '';
+function makeExpressHandler(client, config) {
 
-  if (isPlainObject(config) && isPlainObject(config.serviceContext)) {
+  /**
+   * The Express Error Handler function is an interface for the error handler
+   * stack into the Express architecture.
+   * @function expressErrorHandler
+   * @param {Any} err - a error of some type propagated by the express plugin
+   *  stack
+   * @param {Object} req - an Express request object
+   * @param {Object} res - an Express response object
+   * @param {Function} next - an Express continuation callback
+   * @returns {ErrorMessage} - Returns the ErrorMessage instance
+   */
+  function expressErrorHandler(err, req, res, next) {
+    var ctxService = '';
+    var ctxVersion = '';
 
-    ctxService = config.serviceContext.service;
-    ctxVersion = config.serviceContext.version;
-  }
+    if (isPlainObject(config) && isPlainObject(config.serviceContext)) {
```

Comment:
In general, I would prefer validating an interface as opposed to an implementation. If we were to change the representation of `config` to a function or class that had the appropriate properties still defined this check would begin failing without modifying the interface of the object. If we do begin passing `config` incorrectly or with the wrong values it would be better for unit tests to catch the error than for runtime checks to silence it.